.azure-pipelines/buildAndPackage.yml

```diff
@@ -30,10 +30,37 @@ pr: none
 pool:
   vmImage: windows-latest
 
-steps:
-- template: templates/checkout-and-credscan.yml
-- template: templates/install-java.yml
-- template: templates/secure-files.yml
-- template: templates/build-and-coverage.yml
-- template: templates/publish-artefacts.yml
-- template: templates/alert-failure.yml
+stages:
+- stage: Build-Stage
+  jobs:
+  - job: Build
+    steps:
+    - template: templates/build-stage/checkout-and-credscan.yml
+    - template: templates/build-stage/install-java.yml
+    - template: templates/build-stage/secure-files.yml
+    - template: templates/build-stage/build-and-coverage.yml
+    - template: templates/build-stage/publish-artefacts.yml
+    - template: templates/alert-failure.yml
+- stage: Maven-Preview-Stage
+  jobs: 
+  - job: Maven-Preview
+    steps: 
+    - template: templates/maven-preview-stage/secure-files.yml
+    - template: templates/maven-preview-stage/copy-and-build.yml
+    - template: templates/alert-failure.yml
+- stage: Maven-Release-Stage
+  dependsOn: []
```

Comment:
depends on build, condition on master branch, download artefacts

Reply:
'Download artefacts', as in add a template/step where we download the artefacts? Or as in, we need a condition where this stage only occurs if the artefacts are correctly downloaded?

Reply:
As add a template /step